### ISSUE TITLE
juicefs sync failed with 'multipart: NoSuchUpload'

### ISSUE THREAD

**Issue by lixiaofengfeng** [juicedata/juicefs#3957]
**What happened**:
Error occured when I run command 'juicefs sync --check-new' and it returns as below:
2023/08/02 06:18:04.240278 juicefs[1003] <ERROR>: Failed to copy data of  in 13m16.963721297s: multipart: NoSuchUpload: The specified multipart upload does not exist. The upload ID may be invalid, or the upload may have been aborted or completed.
**What you expected to happen**:
The file should be transferred successfully.
**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**
The file size is about 12.55GB.
**Environment**:
- JuiceFS version (use `juicefs --version`) or Hadoop Java SDK version:
- Cloud provider or hardware configuration running JuiceFS:
- OS (e.g `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Object storage (cloud provider and region, or self maintained):
- Metadata engine info (version, cloud provider managed or self maintained):
- Network connectivity (JuiceFS to metadata engine, JuiceFS to object storage):
- Others:

**Comment by lixiaofengfeng** [comment]
![image](https://github.com/juicedata/juicefs/assets/49725515/9d57bf3a-6cdf-453a-b7cc-c96f969c858f)

**Comment by zhijian-pro** [comment]
This should be the problem of the object storage provider. Please contact the supplier first to check why the `upload id` is invalid.

**Comment by davies** [comment]
@lixiaofengfeng Can you reproduce this issue?

**Comment by lixiaofengfeng** [comment]
> reproduce

yes. I created a gateway through command 'juicefs gateway ......' and there is an endpoint that have access to this gateway. Every command is like 'juicefs sync s3://ak:sk@bucketname.gateway_endpoint/data_path s3://......' .
The command work ok with other storage provider.

**Comment by davies** [comment]
Based the content discussed offline, the issue came from BOS (claimed to be compatible with S3, but may have some issues). We can't reproduce this issue with BOS using the BOS SDK, or with other S3 compatible storage providers.
